File: Hot 100/283. 移动零.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n=nums.size();
        int cnt=0;
        int ind=0;
        for(int i=0;i<n;i++){
            if(nums[i]!=0){
                nums[ind]=nums[i];
                ind++;
                cnt++;
            }
        }
        for(int i=cnt;i<n;i++){
            nums[i]=0;
        }
        return;
    }
};

File: 刷题/Hot 100/283. 移动零.go
func moveZeroes(nums []int)  {
    n := len(nums)
    ind := 0
    cnt := 0
    for i:=0; i<n; i++ {
        if nums[i] != 0 {
            nums[ind]=nums[i]
            ind++
            cnt++
        }
    }
    for i:=cnt; i<n; i++ {
        nums[i]=0
    }
    return
}